src/library/dlgtrackinfo.cpp

```diff
@@ -467,12 +529,37 @@ void DlgTrackInfo::slotBpmTap(double averageLength, int numSamples) {
     }
     double averageBpm = 60.0 * 1000.0 / averageLength;
     // average bpm needs to be truncated for this comparison:
-    if (averageBpm != m_dLastBpm) {
-        m_dLastBpm = averageBpm;
+    if (averageBpm != m_dLastTapedBpm) {
+        m_dLastTapedBpm = averageBpm;
         spinBpm->setValue(averageBpm);
     }
 }
 
+void DlgTrackInfo::slotSpinBpmValueChanged(double value) {
+    if (value <= 0) {
+        m_pBeatsClone.clear();
+        return;
+    }
+
+    if (!m_pBeatsClone) {
+        m_pBeatsClone = BeatFactory::makeBeatGrid(m_pLoadedTrack.data(),
```

Comment:
The same here. We do not know which beat will be aligned after changing the Bpm value. 
Using the cue makes the result reliable.